src/compiler/expressionToTypeNode.ts

```diff
@@ -351,7 +351,7 @@ export function createSyntacticTypeNodeBuilder(options: CompilerOptions, resolve
             }
             else if (prop.name.kind === SyntaxKind.ComputedPropertyName) {
                 const expression = prop.name.expression;
-                if (!isPrimitiveLiteralValue(expression, /*includeBigInt*/ false) && !isEntityNameExpression(expression)) {
+                if (!isPrimitiveLiteralValue(expression, /*includeBigInt*/ false)) {

```

Comment:
There is also code in `typeFromObjectLiteral` that is not effectively unused. `isNonNarrowedBindableName` also probably becomes useless since we can now syntactically determine if the computed property is valid so it can probably be removed from the resolver.